### PR TITLE
Merge async proposal back into sync proposal

### PR DESCRIPTION
**Pull request by rbuckton**
This merges the contents of the Async Explicit Resource Management proposal back into this proposal repository.

### CODE REVIEW

File: spec.emu [hunk 1]
```diff
@@ -3439,16 +4156,17 @@ contributors: Ron Buckton, Ecma International
...
             1. If _message_ is not *undefined*, then
-              1. Let _msg_ be ? ToString(_message_).
-              1. Perform CreateNonEnumerableDataPropertyOrThrow(_O_, *"message"*, _msg_).
-            1. Perform ! CreateNonEnumerableDataPropertyOrThrow(_O_, *"error"*, _error_).
-            1. Perform ! CreateNonEnumerableDataPropertyOrThrow(_O_, *"suppressed"*, _suppressed_).
+              1. Let _messageString_ be ? ToString(_message_).
+              1. Perform CreateNonEnumerableDataPropertyOrThrow(_O_, *"message"*, _messageString_).
+            1. Perform ? InstallErrorCause(_O_, _options_).
+            1. Perform CreateNonEnumerableDataPropertyOrThrow(_O_, *"error"*, _error_).
+            1. Perform CreateNonEnumerableDataPropertyOrThrow(_O_, *"suppressed"*, _suppressed_).
```
Comment (ljharb) [inline spec.emu:4169]: i think maybe this shouldn't be here?

File: spec.emu [hunk 2]
```diff
@@ -3776,6 +4558,202 @@ contributors: Ron Buckton, Ecma International
...
+              Description
+            </th>
+          </tr>
+          <tr>
+            <td>
+              [[AsyncDisposableState]]
+            </td>
+            <td>
+              ~pending~ or ~disposed~
+            </td>
```
Comment (ljharb) [inline spec.emu:4736]: if these only have two values then is there a reason not to have it be a boolean?

File: spec.emu [hunk 3]
```diff
@@ -3776,6 +4558,202 @@ contributors: Ron Buckton, Ecma International
...
+            <td>
+              Governs how a disposable stack will react to incoming calls to its `@@asyncDispose` method.
```
Comment (ljharb) [inline spec.emu:4738]:
```suggestion
              Governs how an async disposable stack will react to incoming calls to its `@@asyncDispose` method.
```

File: spec.emu (hunk 1 above)
Comment (rbuckton) [reply spec.emu:4169]: Yeah, copy paste error. The async proposal's spec hadn't yet been amended for the consensus to remove `cause`.

File: spec.emu (hunk 2 above)
Comment (ljharb) [reply spec.emu:4735]: fair enough, just seems like the spec would be significantly simpler that way

**Comment by syg** [comment]
Is this a mechanical merge of the spec text from the async repo? If so I'm happy to rubberstamp it.

**Comment by rbuckton** [comment]
> Is this a mechanical merge of the spec text from the async repo? If so I'm happy to rubberstamp it.

Essentially, yes. The only difference is where I've either updated algorithm steps to match the current (or at least a more recent) version of ECMA-262, such as recent cleanup of AOs to return `~unused~`, etc.

**Comment by rbuckton** [comment]
> Is this a mechanical merge of the spec text from the async repo? If so I'm happy to rubberstamp it.

Can you clarify if this was meant to indicate approval of the PR? If not, can you review and approve?

**Review by syg — APPROVED** [review]
Stamped. Didn't review in depth, trusting that this is just copy/paste from the split out async repo without editorial or normative changes.

**Pull request merged**